extract/bbox_merge.py

```diff
@@ -0,0 +1,18 @@
+from .coordinate import lower_right_most_coordinate, upper_left_most_coordinate
+from .bbox import BBox
+
+
+def bbox_merge(*bboxes):
+    upper_left_coordinates = []
+    lower_right_coordinates = []
+    for bbox in bboxes:
+        try:
+            upper_left_coordinates.append(bbox.upper_left_coordinate)
+            lower_right_coordinates.append(bbox.lower_right_coordinate)
+        except AttributeError:
```

Comment:
Ugh ugh ugh. ugh! ugh!

I hate this. 

Why is an AttributeError happening here?

I *think* it's because you have a `bbox` which has no `upper_left_coordinate` or possibly no `upper_right_coordinate`, right?

But AttributeError is so generic, and the `except` is so removed from where the error happens...

This is where I'd much rather have a user-defined exception (`except CoordinateMissingException`); or have `upper_left_coordinate` always return a value. If it returned None, you'd end up with some Nones in upper_left_coordinates, which upper_left_most_coordinate can skip over....



So in fact that's what I think I'd do.

````python
class BBox(object):

    def init(ulc = None, lrc = None):
        if not ulc and lrc:
            raise some_exception_about_bboxes_needing_0_or_two_corners
        self.ulc = ulc
        self.lrc = lrc
````

This way it's legal to have a bbox that doesn't have an area...

(yes, I realise this contradicts to some extent my earlier comments about it being ugly to sometimes return one thing and other times return None; but based on this code here, this feels like a time when returning None is actually the preferred choice)